manifests/init.pp

```diff
@@ -664,204 +502,169 @@
+  Optional[String] $package_provider = $puppet::params::package_provider,
+  Optional[Stdlib::Absolutepath] $package_source = $puppet::params::package_source,
+  Integer[0, 65535] $port = $puppet::params::port,
+  Boolean $listen = $puppet::params::listen,
+  Array[String] $listen_to = $puppet::params::listen_to,
+  Boolean $pluginsync = $puppet::params::pluginsync,
+  Boolean $splay = $puppet::params::splay,
+  Pattern[/^\d+[smhdy]?$/] $splaylimit = $puppet::params::splaylimit,
+  Variant[Boolean, Stdlib::Absolutepath] $autosign = $puppet::params::autosign,
+  Array[String] $autosign_entries = $puppet::params::autosign_entries,
+  Pattern[/^[0-9]{3,4}$/] $autosign_mode = $puppet::params::autosign_mode,
+  Optional[String] $autosign_content = $puppet::params::autosign_content,
+  Integer[0] $runinterval = $puppet::params::runinterval,
+  Boolean $usecacheonfailure = $puppet::params::usecacheonfailure,
+  Enum['cron', 'service', 'systemd.timer', 'none'] $runmode = $puppet::params::runmode,
+  Array[Enum['cron', 'service', 'systemd.timer', 'none']] $unavailable_runmodes = $puppet::params::unavailable_runmodes,
+  Optional[String] $cron_cmd = $puppet::params::cron_cmd,
+  Optional[String] $systemd_cmd = $puppet::params::systemd_cmd,
+  Boolean $agent_noop = $puppet::params::agent_noop,
+  Boolean $show_diff = $puppet::params::show_diff,
+  Optional[Stdlib::HTTPUrl] $module_repository = $puppet::params::module_repository,
+  Optional[Integer[0]] $configtimeout = $puppet::params::configtimeout,
+  Optional[Variant[String, Boolean]] $ca_server = $puppet::params::ca_server,
+  Optional[Integer[0, 65535]] $ca_port = $puppet::params::ca_port,
+  Optional[String] $ca_crl_filepath = $puppet::params::ca_crl_filepath,
+  Optional[String] $prerun_command = $puppet::params::prerun_command,
+  Optional[String] $postrun_command = $puppet::params::postrun_command,
+  Array[String] $dns_alt_names = $puppet::params::dns_alt_names,
+  Boolean $use_srv_records = $puppet::params::use_srv_records,
+  Optional[String] $srv_domain = $puppet::params::srv_domain,
+  String $pluginsource = $puppet::params::pluginsource,
+  String $pluginfactsource = $puppet::params::pluginfactsource,
+  Hash[String, Data] $additional_settings = $puppet::params::additional_settings,
+  Hash[String, Data] $agent_additional_settings = $puppet::params::agent_additional_settings,
+  Optional[String] $agent_restart_command = $puppet::params::agent_restart_command,
+  String $classfile = $puppet::params::classfile,
+  String $hiera_config = $puppet::params::hiera_config,
+  String $main_template = $puppet::params::main_template,
+  String $agent_template = $puppet::params::agent_template,
+  String $auth_template = $puppet::params::auth_template,
+  Boolean $allow_any_crl_auth = $puppet::params::allow_any_crl_auth,
+  Array[String] $auth_allowed = $puppet::params::auth_allowed,
+  Array[String] $client_package = $puppet::params::client_package,
```

Comment:
This was a breaking change for me.  `client_package` could be a string *or* an array previously.